### ISSUE TITLE
Unable to receive fax

### ISSUE THREAD

**Issue by aloz** [signalwire/freeswitch#712]
Hello,

Trying to receive faxes with FreeSWITCH (no T.38 support, fax machine -> PSTN -> Voip trunk -> FreeSWITCH) - but no luck: hangup with the status Unexpected message received, or Call terminated prematurely.

Have tested with a lot of fax machines - no luck (dialplan1.txt, freeswitch1.log)

the environment is:
FreeSWITCH Version 1.10.3-release+git~20200511T105455Z~129de34d84~64bit (git 129de34 2020-05-11 10:54:55Z 64bit)
Operating System: Debian GNU/Linux 10 (buster)
Kernel: Linux 4.19.0-9-amd64
Architecture: x86-64

My the second approach was to use Hylafax with mod_spandsp (with the original config.FS per each softmodem instance provided with mod_spandsp to integrate it with FreeSWITCH, to bridge the call to the modem) - no luck as well (dialplan-hylafax,txt, hylafax-log.txt)

On receiving fax with FreeSWITCH - I hear how FS and the fax machine are negotiating. On receiving fax with faxgetty/Hylafax - I hear just single BEEP......BEEP......BEEP......

Please check the configuration and logs attached as a single ZIP file and advice.

Thank you!

p.s. One interesting thing: according the log, _FLOW T.30 Queuing phase C_NON_ECM_RX_ but _ECM status         on_

[fs-fax-no-luck.zip](https://github.com/signalwire/freeswitch/files/4828430/fs-fax-no-luck.zip)

**Comment by mjerris** [comment]
try playing with ecm settings maybe?  what codec?  if anything but g711 don’t expect it to work at all, with g711 and low loss expect maybe 80% working without t.38

**Comment by aloz** [comment]
Thank you - the settings are g711 only. If to play with ecm - if no ecm than even no remote station id is received. Seems to me it was affected by the bad trunk quality, and may be is it possible to improve libspandsp or mod_spandsp to work with it ...

**Comment by bwmarrin** [comment]
@mjerris I'm curious, what's the percentage you'd estimate for t38?  Unrelated to this specific issue, but I following the faxing questions a lot and I'm curious what numbers I should consider "normal".

**Comment by aloz** [comment]
You are right - my investigation shows that this is unable to receive inband T.30 fax even with g711...
Also have read an article in the past - one guy provided the statistics - some percents of success to receive T.30...
Even have setup Asterisk with the latest spandsp library... Without success..,,
So I think it was a miskake to receive T.30 over VOIP
